internal/wire/frame_parser.go

```diff
@@ -82,114 +51,134 @@ func (p *FrameParser) parseNext(b []byte, encLevel protocol.EncryptionLevel, v p
 			continue
 		}
 
-		f, l, err := p.parseFrame(b, typ, encLevel, v)
-		parsed += l
-		if err != nil {
-			return nil, parsed, &qerr.TransportError{
+		frameType := FrameType(typ)
```

Comment:
With the improved validation logic, you can verify that the frame type is valid here.